src/components/Product-card/ProductCard.jsx

```diff
@@ -1,118 +1,65 @@
-import "./productCard.css";
+import React from "react";
 import { VscHeart } from "../../utils/icons";
-import { useFilterData } from "../../contexts/filterData-context";
-import { useLocation } from "react-router-dom";
-import { useEffect } from "react";
-import { getProductClass, getOriginalPrice } from "../../utils/productutilFuncs";
-import { useWishlist } from "../../contexts/wishlist-context";
 import { useCart } from "../../contexts/cart-context";
-import { useProductsGridClass } from "../../contexts/productsGridClass-context";
-
-const ProductCard = () => {
-
-    const location = useLocation();
-
-    const category = location?.state?.category
-
-    const newArrivalCategory = location?.state?.isNewArrival
-
-    const discountedProducts = location?.state?.hasDiscount
+import { useWishlist } from "../../contexts/wishlist-context";
+import { getOriginalPrice } from "../../utils/productutilFuncs";
+import { useNavigate } from "react-router-dom";
 
-    const productOnSale = location?.state?.onSale
+const ProductCard = (product) => {
```

Comment:
You can destructure `product` since its properties are used in multiple places in the component.

Reply:
yes, you are right, but at some places i was needing the product object as a whole, so to avoid the problem that it would have caused i didnt destructure it!!